dll/win32/shell32/utils.cpp

```diff
@@ -99,3 +99,135 @@ SHFindComputer(LPCITEMIDLIST pidlRoot, LPCITEMIDLIST pidlSavedSearch)
 
     return SUCCEEDED(hr);
 }
+
+static VOID Int64ToStr(LONGLONG llValue, LPWSTR pszValue)
+{
+    WCHAR szBuff[40];
+    INT ich = 0;
+#if (WINVER >= 0x0600)
+    BOOL bMinus = (llValue < 0);
+
+    if (bMinus)
+        llValue = -llValue;
+#endif
+
+    do
+    {
+        szBuff[ich++] = (WCHAR)(L'0' + (llValue % 10));
+        llValue /= 10;
+    } while (llValue != 0);
+
+#if (WINVER >= 0x0600)
+    if (bMinus)
+        szBuff[ich++] = '-';
+#endif
+
+    do
+    {
+        --ich;
+        *pszValue++ = szBuff[ich];
+    } while (ich > 0);
+
+    *pszValue = UNICODE_NULL;
+}
+
+#define MAX_NUMSEP 6
+
+static VOID
+Int64GetNumFormat(NUMBERFMTW *pDest, const NUMBERFMTW *pSrc, DWORD dwNumberFlags,
+                  LPWSTR pszDecimal, LPWSTR pszThousand)
```

Comment:
Please add SAL annotations, and provide explicit parameters for the sizes of these pszDecimal and pszThousand buffers, instead of hardcoding them below in that function.

Reply:
SAL annotations added.